### PR TITLE
fix(server): migrate to Express 4

### LINKED ISSUE #2214: Update express version to latest 3.x
If you run `npm shrinkwrap` and run it through the nsp checker you'll see a couple of advisories for Express: https://shrinkwraplint.herokuapp.com/repo/mozilla/fxa-content-server

The newest one is https://nodesecurity.io/advisories/serve-static-xss which was discovered on Saturday.

We're using `"express": "3.17.1",` so we may be able to update to the latest Express 3.x branch with minimal disruptions ([3.20.1](https://github.com/strongloop/express/releases) seems to be latest).

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #2214 

* [x] @zaach
* [x] @shane-tomlinson 
* [x] @pdehaan r?

### CODE REVIEW

**Comment by vladikoff** [comment]
Also see http://expressjs.com/guide/migrating-4.html

**Comment by pdehaan** [comment]
AAWWW YYIIISSSS:
```
Running "validate-shrinkwrap" task
/home/travis/build/mozilla/fxa-content-server/npm-shrinkwrap.json
>> Name  Installed  Patched  Vulnerable Dependency
>> qs      0.6.6     >= 1.x  fxa-content-server > bower > request
>> qs      0.6.6     >= 1.x  fxa-content-server > bower > request
Warning: known vulnerable modules found Used --force, continuing.
Done, but with warnings.
```

**Comment by pdehaan** [comment]
Travis seems to be sad.
I found this, which seems to be the cause of several of the errors, but there may be more:
```
FAIL: #get /i18n/client.json with multiple supported languages (4ms)
AssertionError: expected 'application/json' to equal 'application/json; charset=utf-8'
  at Request.<anonymous>  <tests/server/l10n.js:37:14>
  at Request.<anonymous>  <node_modules/intern-geezer/lib/Test.js:101:15>
  at Request._callback  <node_modules/intern-geezer/lib/Test.js:113:16>
  at Request.self.callback  <node_modules/request/request.js:123:22>
  at Request.emit  <events.js:98:17>
  at Request.<anonymous>  <node_modules/request/request.js:1047:14>
  at Request.emit  <events.js:117:20>
  at IncomingMessage.<anonymous>  <node_modules/request/request.js:998:12>
  at IncomingMessage.emit  <events.js:117:20>
  at <_stream_readable.js:944:16>
```

**Comment by vladikoff** [comment]
> Travis seems to be sad.

Yeap, Should be fixed in latest updated.

**Comment by pdehaan** [comment]
We should totally switch from Express to Hapi! :trollface:

**Comment by pdehaan** [comment]
From Travis:
```
Running: mocha tests
fxa-content-server.legal-templates.WARN: `en` does not exist, trying next best `en-US`
fxa-content-server.legal-templates.WARN: `en` does not exist, trying next best `en-US`
fxa-content-server.legal-templates.WARN: `en` does not exist, trying next best `en-US`
fxa-content-server.legal-templates.WARN: `en` does not exist, trying next best `en-US`
fxa-content-server.legal-templates.WARN: `en` does not exist, trying next best `en-US`
fxa-content-server.legal-templates.WARN: `en` does not exist, trying next best `en-US`
Sending code coverage to coveralls.io
```

Cause for alarm (or something we need to tweak in the env)? I totally don't think it's related to this PR in any way, I just have an annoying "see something, say something" policy.

**Comment by pdehaan** [comment]
I like it. I like it a lot.
@vladikoff: Great work (as always)!

**Comment by vladikoff** [comment]
> Cause for alarm (or something we need to tweak in the env)? I totally don't think it's related to this PR in any way, I just have an annoying "see something, say something" policy.

Not related

**Comment by zaach** [comment]
@pdehaan The templates from the legal repo use `en-US` instead of  `en`, the locale we support for English. We could map legal templates to `en` with a script if that's too weird/confusing.

**Comment by pdehaan** [comment]
@zaach Totally not even worth bothering. It currently works, so it's probably just safest+easiest to ignore me.

File: server/lib/routes/get-client.json.js [hunk 1]
```diff
@@ -24,7 +24,8 @@ module.exports = function (i18n) {
     res.set('Vary', 'accept-language');
 
     // charset must be set on json responses.
-    res.charset = 'utf-8';
+    // set charset for Express 4: https://github.com/strongloop/express/wiki/Migrating-from-3.x-to-4.x#rescharset
+    res.set('content-type', 'application/json; charset=utf-8');
```
Comment (shane-tomlinson) [inline server/lib/routes/get-client.json.js]: @vladikoff - Does this change also need to be made in [get-ver-json.js](https://github.com/mozilla/fxa-content-server/blob/b6e5b00fa4598787abae10eaeb6f74b8c816684b/server/lib/routes/get-ver.json.js#L126)?

**Comment by shane-tomlinson** [comment]
Only one comment made - ver.json is served with the correct char-type without the update, so you decide.

r+ 

Mind if we merge at the beginning of train-35 since it involves an express major version update?

**Comment by jrgm** [comment]
> Mind if we merge at the beginning of train-35 since it involves an express major version update?

A most excellent idea ;-)

**Pull request merged**